ipa-core/src/report/hybrid.rs

```diff
@@ -179,12 +285,70 @@ mod test {
         let enc_report_bytes = oprf_report
             .encrypt(key_id, &key_registry, &mut rng)
             .unwrap();
-        let hybrid_report2 = HybridReport::<BA8, BA3>::from_bytes::<_, _, BA20>(
-            enc_report_bytes.as_slice(),
-            &key_registry,
-        )
-        .unwrap();
+        let enc_report = EncryptedHybridReport {
+            bytes: enc_report_bytes.into(),
+        };
+        let hybrid_report2 = enc_report
+            .decrypt::<_, BA8, BA3, BA20>(&key_registry)
+            .unwrap();
 
         assert_eq!(hybrid_report, hybrid_report2);
     }
+
+    #[test]
+    fn unique_encrypted_hybrid_reports() {
+        #[derive(Clone)]
+        pub struct UniqueByteHolder {
+            bytes: Vec<u8>,
+        }
+
+        impl UniqueByteHolder {
+            pub fn new(size: usize) -> Self {
+                let bytes = generate_random_bytes(size);
+                UniqueByteHolder { bytes }
+            }
+        }
+
+        impl UniqueBytes for UniqueByteHolder {
+            fn unique_bytes(&self) -> Vec<u8> {
+                self.bytes.clone()
+            }
+        }
+
+        let bytes1 = UniqueByteHolder::new(4);
+        let bytes2 = UniqueByteHolder::new(4);
+        let bytes3 = UniqueByteHolder::new(4);
+        let bytes4 = UniqueByteHolder::new(4);
+
+        let mut unique_bytes = UniqueBytesValidator::new(4);
+
+        unique_bytes.check_duplicate(&bytes1).unwrap();
+
+        unique_bytes
+            .check_duplicates(&[bytes2.clone(), bytes3.clone()])
+            .unwrap();
+        let expected_err = unique_bytes.check_duplicate(&bytes2);
+
+        assert!(
+            expected_err.is_err(),
+            "Expected an error, but got Ok instead"
+        );
+
+        expected_err.unwrap_or_else(|err| match err {
+            Error::DuplicateBytes(value) => assert_eq!(value, 3),
+            _ => panic!("Expected DuplicateBytes error, got: {err}"),
+        });
```

Comment:
you could use `assert!(matches!(expected_err, Err(Error::DuplicateBytes())` here

Reply:
ooo that's much nicer